django/contrib/admin/options.py

```diff
@@ -1690,9 +1691,14 @@ def delete_view(self, request, object_id, extra_context=None):
             obj_display = force_text(obj)
             attr = str(to_field) if to_field else opts.pk.attname
             obj_id = obj.serializable_value(attr)
-            self.log_deletion(request, obj, obj_display)
-            self.delete_model(request, obj)
+            obj_clone = copy.copy(obj)
```

Comment:
As you mentioned, the situation could occur when two users are using the admin simultaneously, one deleting the object and the other changing the foreign key. I think redirecting to the intermediate page which informs the user that the object is protected is most helpful in that case.